client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx

```diff
@@ -81,17 +81,25 @@ const CompletingPurchase: Step = function CompletingPurchase( { navigation, flow
 		}
 	};
 	useEffect( () => {
+		let navigateTo = '';
 		if ( flow === 'link-in-bio' ) {
+			navigateTo = 'launchpad';
+
 			completeLinkInBioFlow();
 		} else {
+			navigateTo = 'subscribers';
+
 			setPendingAction( async () => {
 				setProgressTitle( __( 'Completing Purchase' ) );
 				setProgress( 0.3 );
-				await wait( 2000 );
+				await wait( 1000 );
 				setProgress( 1 );
+				await wait( 2000 );
```

Comment:
Shouldn't we just call `submit`/`navigate` here and be happy without passing `navigateTo` around?

Reply:
The problem here is:
```js
case 'completingPurchase':
	return navigate( 'processing', { navigateTo: 'subscribers' } );
case 'subscribers':
	return navigate( 'processing', { navigateTo: 'launchpad' } );
```

We have two different steps that need to show some processing. `navigateTo` is used to determine where to navigate after processing is done!
Tell me if this makes sense for you. For sure there may be a better way for this, but looked like a quick solution for better managing after processing redirection

Reply:
How about we make the pending action return the destination?

In this [line](https://github.com/Automattic/wp-calypso/blob/update/newsletter-flow-pre-checkout/client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx#L44), we can do
```js
const destination = await action();
submit?.( {}, ProcessingResult.SUCCESS, destination );
```

This way, [the pending action](https://github.com/Automattic/wp-calypso/blob/update/newsletter-flow-pre-checkout/client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx#L97) can return the destination.

Reply:
I really like that!